test/common/router/config_impl_test.cc

```diff
@@ -6166,6 +6167,28 @@ TEST_F(RouteEntryMetadataMatchTest, ParsesMetadata) {
   }
 }
 
+class RouteConfigurationDirectResponseBodyTest : public testing::Test, public ConfigImplTestBase {};
+
+TEST_F(RouteConfigurationDirectResponseBodyTest, DirectResponseBodySizeTooLarge) {
```

Comment:
What about a positive test? I'm not sure if an integration test makes sense; if we already support one for direct response, I'm thinking the actual size doesn't make any difference there.

Reply:
Added a case above. And the cases of different data sources combination for setting body content is covered in `RouteMatcherTest.DirectResponse`.